templates/fine-tune-llm_v2/cookbooks/long_context/README.ipynb

```diff
@@ -0,0 +1,115 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Fine-tuning on datasets with long context\n",
+    "**⏱️ Time to complete**: 90 minutes (assuming GCP A2 nodes can be acquired immidiately)\n",
+    "\n",
+    "This guide demonstrates how to prepare a dataset with long context lengths.\n",
+    "Make sure you have gone over the [basic fine-tuning guide](../../README.md) before going over this cookbook.\n",
+    "\n",
+    "In the following example, we filter a hugginface dataset and assemble it into the right format.\n",
+    "By estimating how many tokens an example from the dataset will result in, we can disgard examples that are too long.\n",
+    "You can use this as a template for creating your own datasets.\n",
+    "\n",
+    "> **_NOTE:_** To fine-tune with a context length of 8k tokens and the llama-3-8b.yaml config in this cookbook, we require GCP A2 nodes. For that, you need to instantiate your workspace in a region such as GCP's `us-central1`  due to the relatively high availability. See [GCP's availability info](https://cloud.google.com/compute/docs/gpus/gpu-regions-zones) for more information. If you only want to generate the dataset, you can do that anywhere."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import pandas as pd\n",
+    "from transformers import AutoTokenizer\n",
+    "\n",
+    "# Fill in your personal hugginface token with access to the tokenizer (You can use a similar tokenizer as a work-around)\n",
+    "HHUGGINFACE_TOKEN = \"\"\n",
+    "# The name of the model you want to fine-tune with. We use this only for tokenization so models with the same tokenizer are interoperable here.\n",
+    "MODEL_NAME = \"meta-llama/Meta-Llama-3-8B-Instruct\"\n",
```

Comment:
Llama-3-8B already is 8k context. So what is the point ?

Reply:
The point is that, to users, context extension should be transparent.
Just like it was on endpoints. There is no point in using a 16k dataset if it's transparent to them anyway.

The point of this notebook is to show users how to prepare their dataset if examples are rather long.
8k context length is already quite the long context length and we are still talking about a template here. That is, this is also meant to be an examples that people can just run. It takes about twice as much time to grab the necessary amount of A100s to fine-tune at 16k. And it will also take much longer to fine-tune. That's why I chose 8k after all.